lib/seccomp/CSystemCallFilter_Linux.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#include "seccomp/CSystemCallFilter.h"
+
+#include <core/CLogger.h>
+
+#include <linux/audit.h>
+#include <linux/filter.h>
+#include <sys/prctl.h>
+#include <sys/syscall.h>
+
+
+namespace ml {
+namespace seccomp {
+
+// The old x32 ABI always has bit 30 set in the sys call numbers.
+// The x64 architecture should fail these calls
+unsigned int UPPER_NR_LIMIT = 0x3FFFFFFF;
+
+// Offset to the nr field in struct seccomp_data
+unsigned int SECCOMP_DATA_NR_OFFSET   = 0x00;
+// Offset to the arch field in struct seccomp_data
+unsigned int SECCOMP_DATA_ARCH_OFFSET = 0x04;
+
+// Copied from seccomp.h
+// seccomp.h cannot be included as it was added in Linux kernal 3.17
+// and this must build on older versions.
+#define SECCOMP_MODE_FILTER 2
+#define SECCOMP_RET_ERRNO   0x00050000U
+#define SECCOMP_RET_ALLOW   0x7fff0000U
+#define SECCOMP_RET_DATA    0x0000ffffU
+
+// Added in Linux 3.5
+#ifndef PR_SET_NO_NEW_PRIVS
+#define PR_SET_NO_NEW_PRIVS 38
+#endif
+
+struct sock_filter filter[] = {
```

Comment:
I think this should be `const` and called `FILTER`.  That will probably mean using `const_cast` at the point of use, but our coding style says to prefer making things that are logically constant `const`.